src/sdk/k4a.c

```diff
@@ -526,171 +526,252 @@ static k4a_result_t validate_configuration(k4a_context_t *device, const k4a_devi
+        }
     }
+
     if (K4A_SUCCEEDED(result))
     {
-        result = K4A_RESULT_FROM_BOOL(config->wired_sync_mode >= K4A_WIRED_SYNC_MODE_STANDALONE &&
-                                      config->wired_sync_mode <= K4A_WIRED_SYNC_MODE_SUBORDINATE);
+        if (config->camera_fps != K4A_FRAMES_PER_SECOND_5 && config->camera_fps != K4A_FRAMES_PER_SECOND_15 &&
+            config->camera_fps != K4A_FRAMES_PER_SECOND_30)
+        {
+            result = K4A_RESULT_FAILED;
+            LOG_ERROR("The configured camera fps %d is not supported.", config->camera_fps);
+        }
     }
 
-    if (K4A_SUCCEEDED(result) && (config->wired_sync_mode == K4A_WIRED_SYNC_MODE_SUBORDINATE ||
-                                  config->wired_sync_mode == K4A_WIRED_SYNC_MODE_MASTER))
+    if (K4A_SUCCEEDED(result))
     {
-        bool sync_in_cable_present;
-        bool sync_out_cable_present;
-        result = TRACE_CALL(
-            colormcu_get_external_sync_jack_state(device->colormcu, &sync_in_cable_present, &sync_out_cable_present));
-
-        if (K4A_SUCCEEDED(result) && config->wired_sync_mode == K4A_WIRED_SYNC_MODE_SUBORDINATE)
+        if (config->wired_sync_mode < K4A_WIRED_SYNC_MODE_STANDALONE ||
+            config->wired_sync_mode > K4A_WIRED_SYNC_MODE_SUBORDINATE)
         {
-            result = K4A_RESULT_FROM_BOOL(sync_in_cable_present == true);
+            result = K4A_RESULT_FAILED;
+            LOG_ERROR("The configured wired sync mode %d is not supported.", config->wired_sync_mode);
         }
-        if (K4A_SUCCEEDED(result) && config->wired_sync_mode == K4A_WIRED_SYNC_MODE_MASTER)
+    }
+
+    if (K4A_SUCCEEDED(result))
+    {
+        if (config->wired_sync_mode == K4A_WIRED_SYNC_MODE_SUBORDINATE ||
+            config->wired_sync_mode == K4A_WIRED_SYNC_MODE_MASTER)
         {
-            result = K4A_RESULT_FROM_BOOL(sync_out_cable_present == true);
+            bool sync_in_cable_present;
+            bool sync_out_cable_present;
+
+            result = colormcu_get_external_sync_jack_state(device->colormcu,
+                                                           &sync_in_cable_present,
+                                                           &sync_out_cable_present);
 
-            if (K4A_SUCCEEDED(result) && config->color_resolution == K4A_COLOR_RESOLUTION_OFF)
+            if (K4A_SUCCEEDED(result))
             {
-                LOG_ERROR("Device wired_sync_mode is set to K4A_WIRED_SYNC_MODE_MASTER, so color camera must be used "
-                          "on master device. Color_resolution can not be set to K4A_COLOR_RESOLUTION_OFF.",
-                          0);
-                return K4A_RESULT_FAILED;
+                if (config->wired_sync_mode == K4A_WIRED_SYNC_MODE_SUBORDINATE && !sync_in_cable_present)
+                {
+                    result = K4A_RESULT_FAILED;
+                    LOG_ERROR("Failure to detect presence of sync in cable with wired sync mode %d.",
```

Comment:
```suggestion
                    LOG_ERROR("Failure to detect presence of sync in cable with wired sync mode K4A_WIRED_SYNC_MODE_SUBORDINATE.",
```

This would be a lot more readable than "Failure to detect presence of sync in cable with wired sync mode 2"